[Obs UX] Enable Synthetics on Serverless

justinkambic opened the pull request:
## Summary

Resolves https://github.com/elastic/synthetics-dev/issues/290.

~Creating this PR as a place to experiment with Synthetics running against the Serverless platform. Main goals are to find areas of the codebase that require revision and get as much of Synthetics functioning as possible without access to public locations.~

This adds the necessary config and other features to make Synthetics workable as part of the oblt serverless project. Notably, we aren't including the `xpack.uptime.enabled` flag here, because we are not ready to expose the plugin to production users yet. We're going to enable the plugin on a per-env basis using other means to start, and when we are ready to expose the plugin generally in prod, we will add that flag.

Also adds nav and fixes a few other things that were broken for serverless specifically.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

justinkambic committed:
Begin experimenting with serverless.

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `/oblt-deploy-serverless` : Deploy a serverless Kibana instance using the Observability test environments.
- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)

</p>
</details>

justinkambic commented:
/oblt-deploy-serverless

justinkambic commented:
/oblt-deploy-serverless

justinkambic commented:
/oblt-deploy-serverless

justinkambic commented:
/oblt-deploy-serverless

justinkambic committed:
Include cert paths and manifest location for QA.

justinkambic committed:
Fix nav.

justinkambic committed:
Adjust config to prepare for merge.

Cleanup before review.

Remove unneeded dependency.

Remove legacy uptime setting from config.

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

cauemarcondes reviewed (approved):
Didn't test it, but code LGTM

pgayvallet commented on config/serverless.oblt.yml:
```diff
@@ -34,6 +34,10 @@ xpack.observability.createO11yGenericFeatureId: true
 ## APM Serverless Onboarding flow
 xpack.apm.serverlessOnboarding: true
 
+# Synthetics mTLS cert locations
+xpack.uptime.service.tls.certificate: /mnt/elastic-internal/http-certs/tls.crt
+xpack.uptime.service.tls.key: /mnt/elastic-internal/http-certs/tls.key
```
I'm not sure how to feel about that one. What will be the behavior when running locally an observability project via `--serverless=observability`?. 

Per-project type config files are supposed to be for "generic" (not "env" dependent) configurations. For settings depending on running in MKI, we set them in from the k8s project/kibana controller instead.

justinkambic replied:
> What will be the behavior when running locally an observability project via --serverless=observability?

Nothing should be impacted because these values are only needed when running in a cloud Project. For local development, we either hit a locally-running service in Minikube (which does not use mTLS), or we hit the QA service, which supports basic auth, and is configured in `kibana.dev.yml`.

For QA/Staging/Prod, these paths are the same so no need to configure separate paths. We could, of course, include them via kibana controller but as I understand it we'd be injecting the same key/value for each one in all envs.

shahzad31 commented:
/oblt-deploy-serverless

shahzad31 commented on x-pack/plugins/serverless_observability/public/components/side_navigation/index.tsx:
```diff
@@ -186,6 +186,27 @@ const navigationTree: NavigationTreeDefinition = {
+            {
+              link: 'synthetics:overview',
+              getIsActive: ({ pathNameSerialized, prepend }) => {
+                return pathNameSerialized.startsWith(prepend('/app/synthetics'));
```
```suggestion
                return pathNameSerialized.endsWith(prepend('/app/synthetics'));
```

shahzad31 commented:
Selecting Management was selecting both , suggested a fix

<img width="1719" alt="image" src="https://github.com/elastic/kibana/assets/3505601/ff2e87dc-77c9-4d1d-b502-f3c001275fda">

shahzad31 commented on config/serverless.oblt.yml:
```diff
@@ -34,6 +34,10 @@ xpack.observability.createO11yGenericFeatureId: true
 xpack.apm.serverlessOnboarding: true
 
+# Synthetics mTLS cert locations
+xpack.uptime.service.tls.certificate: /mnt/elastic-internal/http-certs/tls.crt
+xpack.uptime.service.tls.key: /mnt/elastic-internal/http-certs/tls.key
```
@pgayvallet i am kinda surprise that this is working, so in base serverless config we have 
`xpack.uptime.enabled: false`

even though this PR isn't removing that. it works and have enabled that plugin. Not sure how.

shahzad31 reviewed (approved):
LGTM !!
Just need to check if we need to add override the base serverless key 

`xpack.uptime.enabled: false`

justinkambic commented:
> Just need to check if we need to add override the base serverless key

@shahzad31 I'm at ON week this week, so I haven't been paying close attention to GH. Seeing Synthetics running doesn't completely surprise me, because the enabled key is getting injected on a per-env basis now. Omitting the key should result in the plugin being disabled by default, but we're injecting an `enabled: true` key for qa and staging. In production, we do not have the key enabled. My guess is this injected key overrides the existing one defined in the `serverless.yml`.

Once we are ready for Synthetics to be generally available in Serverless, we'll add the `enabled: true` key for all envs and stop injecting it via the controller.

justinkambic commented:
/oblt-deploy-serverless

justinkambic commented:
/oblt-deploy

achyutjhunjhunwala reviewed (approved):
Logs Only changes looks good
